netket/hilbert/abstract_continuous_space_particle.py

```diff
@@ -0,0 +1,78 @@
+from typing import Tuple
+
+import jax.numpy as jnp
+
+from .abstract_hilbert import AbstractHilbert
+
+
+class AbstractParticle(AbstractHilbert):
+    """Abstract class for the Hilbert space of particles
+    in continuous space.
+    This class defines the common interface that
+    can be used to interact with particles defined
+    in continuous space.
+    """
+
+    def __init__(self, N: int, L: Tuple[float, ...], pbc: Tuple[bool, ...]):
+        """
+        Constructs new ``Particles`` given specifications
+         of the continuous space they are defined in.
+
+        Args:
+            N: Number of particles
+            L: spatial extension in each spatial dimension
+            pbc: Whether or not to use periodic boundary
+                conditions for each spatial dimension
+        """
+        self._N = N
+        self._L = L
+        self._pbc = pbc
+
+        if not len(self._L) == len(self._pbc):
+            raise AssertionError(
+                """You need to define boundary conditions for each spatial dimension."""
+            )
+
+        super().__init__()
+
+    @property
+    def size(self) -> int:
+        return self._N * len(self._L)
+
+    @property
+    def N(self) -> int:
+        r"""The number of particles"""
+        return self._N
```

Comment:
It is customary for attributes to be lowercase, with a few exceptions.
But it's better to stick to the customs.

Moreover, i'd go for a more explicit name like `n_particles`